pkg/admission/admission_controller.go

```diff
@@ -524,11 +526,28 @@ func (c *AdmissionController) namespaceMatchesNoLabelList(namespace string) bool
 	return false
 }
 
+// shouldProcessNamespace returns true if the pod in the namespace must be redirected to the
+// YuniKorn scheduler by setting the schedulerName
+// First check is the namespace annotation (tri-state)
+// - if present (0, 1) return the value as boolean
+// - if not present (-1) fallback to matching names based on regexp
 func (c *AdmissionController) shouldProcessNamespace(namespace string) bool {
+	process := c.nsCache.enableYuniKorn(namespace)
+	if process != -1 {
+		return process == 1
```

Comment:
1 / -1 are special values, could you name them? Difficult to reason about them without names.

Reply:
Follow up jira to move to an iota: [YUNIKORN-1652](https://issues.apache.org/jira/browse/YUNIKORN-1653)